### PR TITLE
Add FuncMap to Templates and validate them

### PR DESCRIPTION
**Pull request by tejal29**
Also change all cobra.Run to cobra.RunE so any errors when parsing and executing templates are propagated to ctc_lib.Execute() entry point.

### CODE REVIEW

File: ctc_lib/template_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	}
+	SetExitOnError(false)
+	err := ExecuteE(&testCommand)
+	expectedError := ("can't evaluate field greet in type ctc_lib.TestInterface")
+	if !strings.Contains(err.Error(), expectedError) {
+		t.Errorf("Expected Error to contain: \n %q \nGot:\n %q\n", expectedError, err.Error())
+	}
+}
+
+func TestContainerToolTestOutputWithFuncTemplate(t *testing.T) {
```
Comment (dlorenc) [inline ctc_lib/template_test.go]: Could you add a test with a nil funcmap?

Comment (tejal29) [reply ctc_lib/template_test.go]: done
